lib/index.js

```diff
@@ -3055,6 +3059,254 @@ UFDS.prototype.listResolvers = function listResolvers(region, cb, noCache) {
     this.search(dn, opts, cb, noCache);
 };
 
+/**
+ * AccessKeys related methods
+ */
+
+/**
+ * Adds a new AccessKey to a given user record.
+ *
+ * The AccessKeyId and AccessKeySecret will be generated.
+ *
+ * This method will return you the full key as processed by UFDS.
+ *
+ * @param {Object} user the user record you got from getUser.
+ * @param {String} (Optional) account uuid for a customer sub-user
+ * @param {Function} callback of the form fn(err, key).
+ * @throws {TypeError} on bad input.
+ */
+UFDS.prototype.addAccessKey = function addAccessKey(user, account, cb) {
+    if (typeof (account) === 'function') {
+        cb = account;
+        account = null;
+    }
+    if (typeof (user) !== 'string') {
+        assert.object(user, 'user');
+        if (user.account) {
+            account = user.account;
+        }
+    }
+    assert.func(cb, 'callback');
+    account = workAroundUnderspecification(account, 'account');
+
+    cb = once(cb);
+
+    var self = this;
+
+    function _addAccKey(init_err, usr) {
+        if (init_err) {
+            cb(init_err);
+            return;
+        }
+
+        var entry = {
+            objectclass: 'accesskey',
+            created: Date.now()
+        };
+
+        try {
+            var accessKeyId = crypto.randomBytes(16).toString('hex');
+            var accessKeySecret = crypto.randomBytes(32).toString('hex');
+            entry.accesskeyid = accessKeyId;
+            entry.accesskeysecret = accessKeySecret;
+        } catch (e) {
+            cb(new InternalError(e.message));
+            return;
+        }
+
+        var dn = (account) ?
+            sprintf(SUBUSER_ACCESS_KEY_FMT,
+                entry.accesskeyid, usr.uuid, account) :
+            sprintf(USER_ACCESS_KEY_FMT, entry.accesskeyid, usr.uuid);
+
+        self.add(dn, entry, function (err) {
+            if (err) {
+                cb(translateError(err));
+                return;
+            }
+            self.getAccessKey(usr, entry.accesskeyid, account, cb);
+        });
+    }
+
+    if (typeof (user) === 'object') {
+        _addAccKey(null, user);
+    } else {
+        this.getUser(user, account, _addAccKey);
+    }
+};
+
+
+/**
+ * Retrieves an AccessKey by Id.
+ *
+ * @param {Object} user the object you got back from getUser.
+ * @param {String} AccessKeyId of the AccessKey you want.
+ * @param {String} (Optional) account uuid for a customer sub-user
+ * @param {Function} callback of the form fn(err, key).
+ * @throws {TypeError} on bad input.
+ */
+UFDS.prototype.getAccessKey =
+function getAccessKey(user, id, account, cb, noCache) {
```

Comment:
Any reason why noCache isn't placed before the cb argument? Also, add `noCache` to the docstring comment.

Same goes for following functions.

Reply:
The reason was that `noCache` was added to the existing functions when the library already was in use, and put at the end of the argument list trying to do not break any program using it with the previous method signature.

I find logical here to keep adding it at the same place all the other methods have it.

Adding documentation for the methods allowing it.

Reply:
I don't like it (and I think they should have tried harder previously to handle the optional argument), but since it's already using that pattern elsewhere - feel free to ignore.